src/Common/StringSearcher.h

```diff
@@ -660,18 +667,54 @@ class StringSearcher<true, ASCII> : public StringSearcherBase
         if (needle == needle_end)
             return haystack;
 
-        while (haystack < haystack_end)
-        {
 #ifdef __SSE4_1__
-            if (haystack + n <= haystack_end && pageSafe(haystack))
+        /// Here is the quick path when needle_size is 1. Compare the first and second characters if the needle_size >= 2.
+        if (needle + 1 == needle_end)
+        {
+            while (haystack < haystack_end)
             {
-                /// find first character
-                const auto v_haystack = _mm_loadu_si128(reinterpret_cast<const __m128i *>(haystack));
-                const auto v_against_pattern = _mm_cmpeq_epi8(v_haystack, pattern);
+                if (haystack + n <= haystack_end && pageSafe(haystack))
+                {
+                    const auto v_haystack = _mm_loadu_si128(reinterpret_cast<const __m128i *>(haystack));
```

Comment:
It would be nice to replace `auto` by the actual type (e.g. `_m128i`) because it is usually not immediately clear which return type the intrinsics have.

Reply:
Done. I have also checked the other values which from intrinsics.